hydromt/models/model_mesh.py

```diff
@@ -304,17 +474,76 @@ def write_mesh(self, fn: str = "mesh/mesh.nc", **kwargs) -> None:
             os.makedirs(dirname(_fn))
         self.logger.debug(f"Writing file {fn}")
         ds_out = self.mesh.ugrid.to_dataset()
-        if self.mesh.ugrid.grid.crs is not None:
+        if self.crs is not None:
             # save crs to spatial_ref coordinate
-            ds_out = ds_out.rio.write_crs(self.mesh.ugrid.grid.crs)
+            ds_out = ds_out.rio.write_crs(self.crs)
         ds_out.to_netcdf(_fn, **kwargs)
 
+    # Other mesh properties
+    @property
+    def mesh_grids(self) -> Dict:
```

Comment:
Add type hinting for the dict
```suggestion
    def mesh_grids(self) -> Dict[str, Union[xu.Ugrid1d, xu.Ugrid2d]]:
```